src/components/v5/common/Fields/Datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,180 @@
+    text: cancelButtonText,
+  } = cancelButtonProps || {};
+
+  return (
+    <DatePicker<never, WithRange>
+      portalId={isMobile ? DATEPICKER_PORTAL_ID : undefined}
+      calendarContainer={DatepickerContainer}
+      calendarClassName={styles.wrapper}
+      renderCustomHeader={(props) => (
+        <DatepickerCustomHeader
+          startDate={startDate}
+          endDate={selectsRange ? endDate : null}
+          dateFormat={dateFormat}
+          setStartDate={setStartDate}
+          setEndDate={selectsRange ? setEndDate : undefined}
+          onClose={
+            isMobile
+              ? () => {
+                  resetValues();
+                  calendarRef.current?.setOpen(false);
+                }
+              : undefined
+          }
+          {...props}
+        />
+      )}
+      ref={calendarRef}
+      customInput={
+        selectsRange ? (
+          <DatepickerCustomRangeInput
+            startDate={startDateProp}
+            endDate={endDateProp}
+            dateFormat={dateFormat}
+          />
+        ) : (
+          <DatepickerCustomInput />
+        )
+      }
+      dateFormat={dateFormat}
+      popperClassName={clsx(popperClassName, 'max-w-[20.5rem] !z-[1000]')}
+      renderDayContents={(day) => (
+        <div className="react-datepicker__day-content">{day}</div>
+      )}
+      shouldCloseOnSelect={false}
+      popperModifiers={[
+        ...(popperModifiers || []),
+        {
+          name: 'offset',
+          options: {
+            offset: [10, -38],
+          },
+        },
+      ]}
+      selectsRange={selectsRange}
+      onChange={handleChange}
+      selected={selectsRange ? undefined : startDate}
+      startDate={selectsRange ? startDate : undefined}
+      endDate={selectsRange ? endDate : undefined}
+      onClickOutside={resetValues}
+      {...rest}
+    >
+      {((selectsRange && startDate && endDate) ||
+        (!selectsRange && startDate)) && (
+        <div className="p-4 border-t border-t-gray-200 w-full flex items-center justify-between gap-2">
+          <Button
+            {...cancelButtonProps}
+            mode={cancelButtonMode || ('primaryOutline' as ButtonMode)}
+            onClick={(e) => {
+              resetValues();
+              calendarRef.current?.setOpen(false);
+              cancelButtonOnClick?.(e);
+            }}
+            text={cancelButtonText || formatText({ id: 'button.cancel' })}
+            className="flex-grow"
+          />
+          <Button
+            {...applyButtonProps}
+            disabled={
+              !startDate ||
+              (!endDate && !!selectsRange) ||
+              (!!endDate && !!startDate && isAfter(startDate, endDate))
+            }
+            mode={applyButtonMode || ('primarySolid' as ButtonMode)}
+            onClick={(e) => {
+              const onChangeArg = (
+                selectsRange ? [startDate, endDate] : startDate
+              ) as Parameters<typeof onChange>[0];
```

Comment:
If `selectsRange` is `false`, then `onChangeArg` will not be a tuple, so why are we accessing it's first index value in that case ?

![Screenshot from 2024-02-16 19-13-42](https://github.com/JoinColony/colonyCDapp/assets/1193222/3c3fb7fb-beb2-4e49-a515-b6484eecd6ba)